src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -333,10 +336,61 @@ JNI_METHOD(jlong, getDevicePointer)(JNIEnv * env, jobject self, jlong handle, jl
 
     GetCHIPDevice(env, handle, deviceId, &chipDevice);
 
-    static_assert(sizeof(jlong) >= sizeof(void *), "Need to store a pointer in a java handle");
+    static_assert(sizeof(jlong) >= sizeof(void *), "Need to store a pointer in a Java handle");
     return reinterpret_cast<jlong>(chipDevice);
 }
 
+JNI_METHOD(void, getConnectedDevicePointer)(JNIEnv * env, jobject self, jlong handle, jlong nodeId, jobject callback)
+{
+    StackLockGuard lock(JniReferences::GetInstance().GetStackLock());
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    Callback::Callback<OnDeviceConnected> connectedCallback(OnDeviceConnectedFn, callback);
+    Callback::Callback<OnDeviceConnectionFailure> failureCallback(OnDeviceConnectionFailureFn, callback);
+
+    wrapper->Controller()->GetConnectedDevice(nodeId, &connectedCallback, &failureCallback);
```

Comment:
Is it intended that this only works if the connection completes synchronously? The callbacks are dropped at the closing brace...